high_school/templates/high_school/index.html

```diff
@@ -61,7 +62,134 @@
+
+            function range(start, end) {
+                return Array.from(Array(end - start + 1), (_, i) => i + start);
+            }
+
+            let sideWidth = maxLength < 9 ? 1 : 2;
+            let leftWidth = (maxLength - sideWidth * 2 - 3) >> 1;
+            let rightWidth = (maxLength - sideWidth * 2 - 2) >> 1;
+            if (totalPages <= maxLength) {
+                // no breaks in list
+                return range(1, totalPages);
+            }
+            if (page <= maxLength - sideWidth - 1 - rightWidth) {
+                // no break on left of page
+                return range(1, maxLength - sideWidth - 1)
+                    .concat([0])
+                    .concat(range(totalPages - sideWidth + 1, totalPages));
+            }
+            if (page >= totalPages - sideWidth - 1 - rightWidth) {
+                // no break on right of page
+                return range(1, sideWidth)
+                    .concat([0])
+                    .concat(range(totalPages - sideWidth - 1 - rightWidth - leftWidth, totalPages));
+            }
+            // Breaks on both sides
+            return range(1, sideWidth)
+                .concat([0])
+                .concat(range(page - leftWidth, page + rightWidth))
+                .concat([0])
+                .concat(range(totalPages - sideWidth + 1, totalPages));
+        }
+
+        $(function () {
+            let totalPages = {{ paginator.num_pages }};
+            let paginationSize = 7;
+            let currentPage;
+
+            function showPage(whichPage) {
+                if (whichPage < 1 || whichPage > totalPages) return false;
+                currentPage = whichPage;
+                $(".pagination li").slice(1, -1).remove();
+                getPageList(totalPages, currentPage, paginationSize).forEach(item => {
+                    $("<li>").addClass("page-item")
+                        .addClass(item ? "" : "disabled")
+                        .toggleClass("active", item === currentPage).append(
+                        $("<a>").addClass("page-link").attr({
+                            {% with request.get_full_path as querystring  %}
+                                {% if 'query' in querystring or 'loc_bx' in querystring or 'loc_bk' in querystring or 'loc_mn' in querystring or 'loc_qn' in querystring or 'loc_si' in querystring %}
+                                    {% if 'page' in querystring %}
+                                        href: item ? "{{ querystring|slice:":-7" }}&page=" + item : "javascript.void(0)"
+                                    {% else %}
+                                        href: item ? "{{ querystring }}&page=" + item : "javascript.void(0)"
+                                    {% endif %}
+                                {% else %}
+                                    href: item ? "?page=" + item : "javascript.void(0)"
+                                {% endif %}
+                            {% endwith %}
```

Comment:
I think this section causes you to have a long link if you have a filter on and click on page numbers on the bottom. EX: http://127.0.0.1:8000/dashboard/all_schools/27Q351/?query=&amp;amp;amp;amp;amp;amp;loc_qn=on&amp;amp;amp;amp;amp;&amp;amp;amp;amp;&amp;amp;amp;&amp;amp;&amp;&page=14